src/controllers/movies.js

```diff
@@ -0,0 +1,64 @@
+const { Prisma } = require("@prisma/client")
+const prisma = require('../utils/prisma')
+
+const getAllMovies = async (req, res) => {
+    const movies = await prisma.movie.findMany({
+        include: {
+            screenings: true
+        }
+    })
+    return res.status(200).json({movies})
+}
+
+const createMovie = async (req, res) => {
+    const { title, runtimeMins } = req.body
+    const movie = await prisma.movie.create({
+        data: {
+            title,
+            runtimeMins
+        },
+        include: {
+            screenings: true
+        }
+    })
+    return res.status(201).json({ movie })
+}
+
+const getMovieById = async (req, res) => {
+    const id = Number(req.params.id)
+    const movie = await prisma.movie.findUnique({
+        where: {
+            id: id
+        },
+        include: {
+            screenings: true
+        }
+    })
+    return res.status(200).json({ movie })
+}
+
+const updateMovie = async (req, res) => {
+    const id = Number(req.params.id)
+    const {title, runtimeMins} = req.body
+
+    const movie = await prisma.movie.update({
+        where: {
+            id: id
+        },
+        data: {
+            title,
+            runtimeMins
+        },
+        include: {
+            screenings: true
+        }
+    })
+    return res.status(201).json({ movie })
+}
+
```

Comment:
All looks solid, should give you a pretty solid base for the extensions if you choose to continue on to them.

When you get in to adding error checking, and increasing the complexity of the code, it would be worth researching how to split it down in to a controller and domain file structure. Anything that directly interacts with your database, such as the "prisma.movie.update" on line 44 or "prisma.movie.findUnique" on line 29 would be in your domain file. This then passes whatever it receives back to the controller, which then either handles the response or manages errors however you've told it to.